app/controllers/mentoring/discussions_controller.rb

```diff
@@ -17,6 +17,6 @@ def use_mentor_discussion
     @mentor_discussion = Mentor::Discussion.find_by!(uuid: params[:id])
 
     # TODO: (Required) Do we want an unauthorised page here?
-    redirect_to mentoring_path unless @mentor_discussion.mentor_id == current_user.id
+    redirect_to mentoring_path unless @mentor_discussion.viewable_by_mentor?(current_user)
```

Comment:
I've created a new `viewable_by_mentor?` method on the `Discussion` class, as I could not re-use the `viewable_by?` method as that would also return `true` if the current user was the student. I'm not entirely happy with the new method's name though. Let me know what you think.